searchlib/src/vespa/searchlib/attribute/singleboolattribute.h

```diff
@@ -0,0 +1,102 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "integerbase.h"
+#include <vespa/searchlib/common/growablebitvector.h>
+#include <limits>
+
+namespace search {
+
+class SingleBoolAttribute : public IntegerAttributeTemplate<int8_t>
+{
+public:
+    SingleBoolAttribute(const vespalib::string & baseFileName, const search::GrowStrategy & grow);
+    ~SingleBoolAttribute() override;
+
+    void onCommit() override;
+    bool addDoc(DocId & doc) override;
+    void onAddDocs(DocId docIdLimit) override;
+    void onUpdateStat() override;
+    bool onLoad() override;
+    void onSave(IAttributeSaveTarget &saveTarget) override;
+    void clearDocs(DocId lidLow, DocId lidLimit) override;
+    void onShrinkLidSpace() override;
+    void removeOldGenerations(generation_t firstUsed) override;
+    void onGenerationChange(generation_t generation) override;
+    uint64_t getEstimatedSaveByteSize() const override;
+
+    SearchContext::UP
+    getSearch(std::unique_ptr<QueryTermSimple> term, const attribute::SearchContextParams & params) const override;
+
+    uint32_t getValueCount(DocId doc) const override {
```

Comment:
Please move implementation of most virtual method to the .cpp file.

Reply:
This is a pattern used in all numeric variants. I think we should look at them in the same context.